addons/resource/models/resource.py

```diff
@@ -383,7 +387,7 @@ def _get_day_work_intervals(self, day_date, start_time=None, end_time=None, comp
             to_naive_utc(interval[1], self.env.user),
             interval[2]) for interval in working_intervals]
 
-    def _get_day_leave_intervals(self, day_date, start_time, end_time, resource_id):
+    def _get_day_leave_intervals(self, day_date, start_time, end_time, resource_id, time_type='leave'):
```

Comment:
Same as before, I would pass a domain-like parameter.

Note: all callers will have to be updated, I will not put a "same" comment on all method definition :p .